File: apps/api/src/domain/repositories/repositories.ts
// Capa DOMAIN: contratos de repositorios (interfaces). Las implementaciones viven en
// infrastructure/ (Prisma). Los casos de uso dependen de estas interfaces, no de Prisma.

import type { UserEntity } from "../entities/user.entity.js";

export interface UserRepository {
  findById(id: string): Promise<UserEntity | null>;
  findByEmail(email: string): Promise<UserEntity | null>;
  findByPhone(phone: string): Promise<UserEntity | null>;
  create(data: { email?: string; phone?: string; displayName: string; passwordHash?: string }): Promise<UserEntity>;
}

import type { MemberRole } from "@blu-ia/shared";

export interface ProjectRepository {
  findMembership(projectId: string, userId: string): Promise<{ role: MemberRole } | null>;
}

File: apps/api/internal/domain/repositories/repositories.go
package repositories

import (
	"context"

	"github.com/blu-ia/api/internal/domain/entities"
	"github.com/blu-ia/api/internal/shared"
)

type UserRepository interface {
	FindByID(ctx context.Context, id string) (*entities.User, error)
	FindByEmail(ctx context.Context, email string) (*entities.User, error)
	FindByPhone(ctx context.Context, phone string) (*entities.User, error)
	Create(ctx context.Context, data CreateUserData) (*entities.User, error)
}

type CreateUserData struct {
	Email        *string
	Phone        *string
	DisplayName  string
	PasswordHash *string
}

type ProjectRepository interface {
	FindMembership(ctx context.Context, projectID, userID string) (*ProjectMembership, error)
}

type ProjectMembership struct {
	Role shared.MemberRole `json:"role"`
}
